### DISCUSSION TITLE
Clipboard Command Binding (WPF migration)

### DISCUSSION THREAD

**Discussion by AlexeiScherbakov** [AvaloniaUI/Avalonia#12636] [Q&A]
In WPF I can make ViewModels implement interface like:

```
public interface IWithName
{
    string Name{get;}
}
```

Then define static command
```
public static class ViewsClipboardCommands
{
    public static readonly RoutedCommand CopyNameToClipboard = new RoutedCommand();

    public static void CopyNameToClipboardEventHandler(object sender, RoutedEventArgs e)
    {
        var element = (FrameworkElement) e.OriginalSource;
        var dataItem = element.DataContext as IObjectWithName;
        if (dataItem is null)
        {
            return;
        }
        // clipboard Name copy logic
    }

    public static void CopyNameToClipboardCanExecuteEventHandler(object sender, CanExecuteRoutedEventArgs e)
    {
        var element = (FrameworkElement) e.OriginalSource;
        e.CanExecute =element.DataContext is IObjectWithName
    }

    public static void RegisterView(UIElement uiElement)
    {
        uiElement.CommandBindings.Add(new CommandBinding(CopyUrlToClipboard, CopyUrlToClipboardEventHandler,CopyUrlToClipboardCanExecuteEventHandler));
    }
}
```
In WPF this will enable logic of copying clipboard content to all views which are implementing IWithName interface by assigning ViewsClipboardCommands.CopyNameToClipboard to menu item. If RegisterView is placed in base window class - it will enable Name clipboard copy for entire application.

But when I look in Avalonia Clipboard commands (https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml) I saw that Avalonia using different mechanism for commands.

How I can achieve the same effect in Avalonia without adding commands in each viewmodel classes(there are also IWithId and many other interfaces with clipboard copy & paste logic , so I cannot add it to base class)?

**Comment by timunie** [comment]
I think something like this can work:

```cs
public static class AppCommands 
{
   // RelayCommand is from CommunityToolkit.MVVM ... Can also be ReactiveComamnd or whatever
   public static RelayCommand MyCommand {get} = new ...
}
```

Usage 
```xml
<Button Command="{x:Static my:AppCommands.MyCommand}" CommandParameter="{Binding MyParamter}" />
```

**Comment by wosiyourfather** [comment]
1
